streams/actions/src/main/java/com/hortonworks/streamline/streams/actions/topology/state/TopologyStates.java

```diff
@@ -0,0 +1,199 @@
+/**
+ * Copyright 2017 Hortonworks.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+
+ *   http://www.apache.org/licenses/LICENSE-2.0
+
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+package com.hortonworks.streamline.streams.actions.topology.state;
+
+import com.hortonworks.streamline.streams.actions.TopologyActions;
+import com.hortonworks.streamline.streams.catalog.CatalogToLayoutConverter;
+import com.hortonworks.streamline.streams.catalog.Topology;
+import com.hortonworks.streamline.streams.layout.component.TopologyDag;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The class captures the different states a 'topology' can be in and the state transitions.
+ */
+public final class TopologyStates {
```

Comment:
Its pretty clear if you go through the code for the different states below. Each state has a set of valid actions and for each action we move to the appropriate next state.

We can put the different states and transitions in the javadoc, but it will get out of sync with the code soon. So its better we retain it in the code itself since its pretty straightforward here.